h2o-genmodel/src/main/java/hex/genmodel/algos/targetencoder/TargetEncoderMojoReader.java

```diff
@@ -0,0 +1,101 @@
+package hex.genmodel.algos.targetencoder;
+
+import hex.genmodel.ModelMojoReader;
+import hex.genmodel.algos.word2vec.Word2VecMojoModel;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TargetEncoderMojoReader extends ModelMojoReader<TargetEncoderMojoModel> {
+
+  @Override
+  public String getModelName() {
+    return "TargetEncoder";
+  }
+
+  @Override
+  protected void readModelData() throws IOException {
+    _model._targetEncodingMap = parseTargetEncodingMap("feature_engineering/target_encoding.ini");
+  }
+
+  @Override
+  protected TargetEncoderMojoModel makeModel(String[] columns, String[][] domains, String responseColumn) {
+    return new TargetEncoderMojoModel(columns, domains, responseColumn);
+  }
+
+  public Map<String, Map<String, int[]>> parseTargetEncodingMap(String pathToSource) throws IOException {
+    Map<String, Map<String, int[]>> encodingMap = null;
+
+    if(getMojoReaderBackend().exists(pathToSource)) {
+      BufferedReader source = getMojoReaderBackend().getTextFile(pathToSource);
+
+      encodingMap = new HashMap<>();
+      Map<String, int[]> encodingsForColumn = null;
+      String sectionName = null;
+      try {
+        String line;
+
+        while (true) {
+          line = source.readLine();
+          if (line == null) { // EOF
+            encodingMap.put(sectionName, encodingsForColumn);
+            break;
+          }
+          line = line.trim();
+          if (sectionName == null) {
+            sectionName = matchNewSection(line);
+            encodingsForColumn = new HashMap<>();
+          } else {
+            String matchResult = matchNewSection(line);
+            if (matchResult != null) {
+              encodingMap.put(sectionName, encodingsForColumn);
+              encodingsForColumn = new HashMap<>();
+              sectionName = matchResult;
+              continue;
+            }
+
+            String[] res = line.split("\\s*=\\s*", 2);
+            int[] numAndDenom = processNumeratorAndDenominator(res[1].split(" "));
+            encodingsForColumn.put(res[0], numAndDenom);
+          }
+        }
+        source.close();
```

Comment:
This line seems to be duplicated (see the `finally` statement).

Reply:
@mn-mikke Please, read through all history of comments when you are doing a review. I have to  answer the same questions. See https://github.com/h2oai/h2o-3/pull/3282#discussion_r270321640 and take a the look at blame in ModelMojoReader. I already asked question about this.... as I don't see either why we should have two calls to `.close()`